AI lint skill (#690)
theforeman/foremanctl: .agents/skills/lint/SKILL.md

---
name: lint
description: >-
  Run linters after finishing a change. Use when any YAML and code files have been changed.
---

# Lint

Run project code checks.

## Canonical docs

- [AGENTS.md](../../../AGENTS.md) — common lint commands
- [Playbooks and Roles](../../../docs/developer/playbooks-and-roles.md) — secrets, OAuth, naming
- Custom rules: `.ansible-lint-rules/` (source of truth for enforceable constraints)

> - Config: `.ansible-lint`

Reviewer: These are files relative to the root while the above are relative to this file. Feels a bit inconsistent to me.

## Workflow

1. Set up virtual environment
2. Ansible lint
3. Ruff

### Set up virtual environment

```shell
./setup-environment
source .venv/bin/activate
```

### Ansible lint

```shell
cd src && ansible-lint

> cd development && ansible-lint

Reviewer: Does this work as well?
```suggestion
ansible-lint src development
```

Reviewer 2: No, because the ansible.cfg is within each of those respectively and has relative configuration inside it.

---
name: lint
description: >-

>   Run linters after finishing a change. Use when any code (except *.md files) has been changed.

Reviewer: I think YAML files may not be considered code, but here they are relevant. Is that important to call out?

Author: > Is that important to call out?

That's a good question. My approach was "better safe than sorry", but I can remove it.

Reviewer: Perhaps this is good enough?
```suggestion
  Run linters after finishing a change. Use when any YAML and code files have been changed.
```

---

# Lint

Run project code checks.

## Canonical docs

- [AGENTS.md](../../../AGENTS.md) — common lint commands
- [Playbooks and Roles](../../../docs/developer/playbooks-and-roles.md) — secrets, OAuth, naming
- Custom rules: `.ansible-lint-rules/` (source of truth for enforceable constraints)
- Config: `.ansible-lint`

## Workflow

- Ansible lint
- Ruff

### Run `ansible-lint`

```shell
cd src && ansible-lint

> cd development && ansible-lint

Reviewer: Pretty sure this also needs the virtual environment.

Author: Ok, I will add it as a first step.
